### PR TITLE
Fixed FileWather issue on deletion of subfolder

### PR DESCRIPTION
**Pull request by arvindgh**
if the folder is deleted(shift+del) by the user from the file system and not from Ginger StopFileWatcher for that folder

### CODE REVIEW

File: Ginger/GingerCoreCommon/Repository/RepositoryFolder.cs [hunk 1]
```diff
@@ -553,6 +553,22 @@ private void HandleDirecortyChange(FileSystemEventArgs e)
...
                 case WatcherChangeTypes.Changed:
                     // change happen when new file is added, for now we can ignore, as we will get also file added event
+
+                    //if the folder is deleted(shift+del) by the user from the file system and not from Ginger
+                    try
+                    {
+                        Directory.GetFiles(e.FullPath);
+                    }
+                    catch (Exception ex)
+                    {
```
Comment (JinendraGhodke) [inline Ginger/GingerCoreCommon/Repository/RepositoryFolder.cs:563]: can not rely on exceptions and delete folders. Please explore better solution

**Comment by ambarkhan** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 1
           
```


See the complete overview on [Codacy](https://app.codacy.com/manual/Ginger-Automation/Ginger/pullRequest?prid=5863082&bid=19193419)

File: Ginger/GingerCoreCommon/Repository/RepositoryFolder.cs [hunk 2]
```diff
@@ -553,9 +553,21 @@ private void HandleDirecortyChange(FileSystemEventArgs e)
...
+                        RepositoryFolder<T> rf = GetSubFolder(fn);
+                        rf.StopFileWatcherRecursive();
+                    }
                     break;
                 case WatcherChangeTypes.Deleted:
                     RepositoryFolder<T> sf2 = GetSubFolder(fn);
+                    //sf2.StopFileWatcherRecursive();
```
Comment (ambarkhan) [inline Ginger/GingerCoreCommon/Repository/RepositoryFolder.cs:570]: Codacy found an issue: [Remove this commented out code.](https://app.codacy.com/manual/Ginger-Automation/Ginger/pullRequest?prid=5863082)

**Comment by arvindgh** [comment]
No more relevant

**Pull request closed without merging**